### PR TITLE
test: recreate sparse pipeline test for 2.0

### PR DESCRIPTION
**Pull request by Kelton8Z**

### CODE REVIEW

File: tests/integration/sparse_pipeline/test_sparse_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @requests(on='index')
+    def encode(self, docs: 'DocumentArray', *args, **kwargs) -> Any:
+        self.docs.extend(docs)
+        for i, doc in enumerate(self.docs):
+            doc.embedding = sparse.csr_matrix(doc.content)
+            self.vectors[doc.id] = doc.embedding.getrow(i)
+
+    @requests(on='search')
+    def query(self, parameters, *args, **kwargs):
```
Comment (JoanFM) [inline tests/integration/sparse_pipeline/test_sparse_pipeline.py:45]: there is no encoding step at query time?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/integration/sparse_pipeline/test_sparse_pipeline.py (hunk 1 above)
Comment (Kelton8Z) [reply tests/integration/sparse_pipeline/test_sparse_pipeline.py:45]: no longer needed per our discussion ?

File: tests/integration/sparse_pipeline/test_sparse_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        top_k = parameters['top_k']
+        doc = parameters['doc']
+        distances = [item for item in range(0, min(top_k, len(self.docs)))]
+        return [self.docs[:top_k]], np.array([distances])
+
+
+def test_sparse_pipeline(mocker, docs_to_index):
+    def validate(response):
+        assert len(response.docs) == 10
```
Comment (bwanglzu) [inline tests/integration/sparse_pipeline/test_sparse_pipeline.py:54]: does it work for 2.0? I think it should be `response.data.docs`

File: tests/integration/sparse_pipeline/test_sparse_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                assert match.id == docs_to_index[i].id
+                assert isinstance(match.embedding, sparse.coo_matrix)
+
+    f = Flow().add(uses=DummyCSRSparseIndexEncoder)
+
+    mock = mocker.Mock()
+    error_mock = mocker.Mock()
+
+    with f:
+        f.post(
```
Comment (bwanglzu) [inline tests/integration/sparse_pipeline/test_sparse_pipeline.py:66]: we could use `f.index` and remove `on`, checkout 2.0 clean code.

File: tests/integration/sparse_pipeline/test_sparse_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            on='index',
+            inputs=docs_to_index,
+            on_done=mock,
+        )
+        f.post(
```
Comment (bwanglzu) [inline tests/integration/sparse_pipeline/test_sparse_pipeline.py:71]: same `f.search`

**Review by bwanglzu — CHANGES_REQUESTED** [review]

File: tests/integration/sparse_pipeline/test_sparse_pipeline.py (hunk 2 above)
Comment (Kelton8Z) [reply tests/integration/sparse_pipeline/test_sparse_pipeline.py]: Good point

**Review by bwanglzu — DISMISSED** [review]
LGTM!

**Pull request merged**